tests/models/mixins.py

```diff
@@ -213,6 +213,48 @@ def test_dataloader(self):
         return self._dataloader(train=False)
 
 
+class CustomInfDataloader:
```

Comment:
I would rather create complet Dataloader so it is easier to undestand... what about?
```
class CustomInfDataloader:

    def __init__(self, dataset, batch_size, shuffle):
        self.dataset = dataset
        self.batch_size = batch_size
        self.shuffle = shuffle

    def __iter__(self):
        idxs = []
        while True:
            if len(idxs) < self.batch_size:
                idxs = range(len(self.dataset))
                if self.shuffle:
                    np.random.shuffle(idxs)
            batch = [self.dataset[idx] for idx in idxs[:self.batch_size]]
            yield batch
            idxs = idxs[len(batch):]
```

Reply:
`torch.DataLoader` does quite a bit more than this (e.g. collate functions, samplers, etc.) so it is probably better to extend rather than rewrite it - also we don't really have access to the dataset when this is created, only the dataloader

Reply:
@Borda we generally want to avoid duplicating torch functionality. Otherwise the project scope will blow up quickly,

Reply:
I  do agree, I just found this construction quite difficult to follow...